### DISCUSSION TITLE
Basic URL parser?

### DISCUSSION THREAD

**Discussion by ddevienne** [uNetworking/uWebSockets#1683] [Q&A]
Hi. Obviously _uWebSockets_ understands and internally parses URLs.
But it does not expose a formal URL parser, or URL type, to validate a URL or give access to its components / parts.
Is this something outside the library's goal and we are supposed to use something external like [Boost.URL][1] or [`QUrl`][2]?
Or can we expect something in the near future? Thanks, --DD

[1]: https://www.boost.org/doc/libs/1_83_0/libs/url/doc/html/index.html
[2]: https://doc.qt.io/qt-6/qurl.html

**Comment by uNetworkingAB** [comment]
you have getMethod, getQuery(key), getQuery(), getParameter(index) and maybe some more if I forgot. it's in the docs

**Reply by ddevienne** [reply]
Those are methods of `HttpRequest`. I don't have a request, I want to validate and manipulate URLs in general, from strings.
There's no constructor for `HttpRequest` to _abuse_ it for its URL-related methods. And it's created via `HttpParser` who's
sole public method does not seem amenable to _abuse_ either. Is that clearer?
